doc/CMakeLists.txt

```diff
@@ -46,65 +46,65 @@ IF(WITH_APIDOC)
     ENDIF(GENERATE_QHP)
 
     SET(DOXYGEN_INCLUDE_PATH
-      ${CMAKE_SOURCE_DIR}/src/core
+      ${CMAKE_CURRENT_SOURCE_DIR}/../src/core
     )
     STRING(REPLACE ";" " " DOXYGEN_INCLUDE_PATH "${DOXYGEN_INCLUDE_PATH}")
 
     SET(DOXYGEN_INPUT
-      ${CMAKE_SOURCE_DIR}/doc
-      ${CMAKE_SOURCE_DIR}/src/core
-      ${CMAKE_SOURCE_DIR}/src/core/annotations
-      ${CMAKE_SOURCE_DIR}/src/core/auth
-      ${CMAKE_SOURCE_DIR}/src/core/composer
-      ${CMAKE_SOURCE_DIR}/src/core/diagram
-      ${CMAKE_SOURCE_DIR}/src/core/dxf
-      ${CMAKE_SOURCE_DIR}/src/core/effects
-      ${CMAKE_SOURCE_DIR}/src/core/fieldformatter
-      ${CMAKE_SOURCE_DIR}/src/core/geometry
-      ${CMAKE_SOURCE_DIR}/src/core/gps
-      ${CMAKE_SOURCE_DIR}/src/core/layertree
-      ${CMAKE_SOURCE_DIR}/src/core/metadata
-      ${CMAKE_SOURCE_DIR}/src/core/pal
-      ${CMAKE_SOURCE_DIR}/src/core/processing
-      ${CMAKE_SOURCE_DIR}/src/core/providers
-      ${CMAKE_SOURCE_DIR}/src/core/providers/memory
-      ${CMAKE_SOURCE_DIR}/src/core/raster
-      ${CMAKE_SOURCE_DIR}/src/core/scalebar
-      ${CMAKE_SOURCE_DIR}/src/core/symbology-ng
-      ${CMAKE_SOURCE_DIR}/src/gui
-      ${CMAKE_SOURCE_DIR}/src/gui/auth
-      ${CMAKE_SOURCE_DIR}/src/gui/attributetable
-      ${CMAKE_SOURCE_DIR}/src/gui/editorwidgets
-      ${CMAKE_SOURCE_DIR}/src/gui/editorwidgets/core
-      ${CMAKE_SOURCE_DIR}/src/gui/effects
-      ${CMAKE_SOURCE_DIR}/src/gui/layertree
-      ${CMAKE_SOURCE_DIR}/src/gui/locator
-      ${CMAKE_SOURCE_DIR}/src/gui/raster
-      ${CMAKE_SOURCE_DIR}/src/gui/symbology-ng
-      ${CMAKE_SOURCE_DIR}/src/analysis
-      ${CMAKE_SOURCE_DIR}/src/analysis/interpolation
-      ${CMAKE_SOURCE_DIR}/src/analysis/network
-      ${CMAKE_SOURCE_DIR}/src/analysis/openstreetmap
-      ${CMAKE_SOURCE_DIR}/src/analysis/raster
-      ${CMAKE_SOURCE_DIR}/src/analysis/vector
-      ${CMAKE_SOURCE_DIR}/src/plugins
+      ${CMAKE_CURRENT_SOURCE_DIR}/../doc
```

Comment:
The issue is that this file is included from `.ci/travis/code_layout/CMakeLists.txt` hence the cmake top source directory is not the same.
In general, I would prefer to use the relative path for this reason, it allows more flexibilty.
What do you think?